### PR TITLE
Also array_slice wide email matches of local users

### LINKED ISSUE #8454: Sharees endpoint doesn't respect limit for ldap users
### Steps to reproduce
1. Setup ldap
2. Query the files_sharing endpoint with a user limit of 1:
`curl 'http://admin:admin@localhost:8140/ocs/v2.php/apps/files_sharing/api/v1/sharees?format=json&search=&perPage=1&itemType=emails' -H 'OCS-APIREQUEST: true'`

### Expected behaviour
The result set in ocs.data.users should only contain 1 entry

### Actual behaviour
The result set in ocs.data.users contains all ldap users

Came up in https://github.com/nextcloud/deck/issues/422 but I can reproduce that in the files app, where searching for a user to share with takes ages with 1000 users in ldap.

@nextcloud/sharing @nextcloud/ldap

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #8454

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]
Works as expected now :+1: Thanks @nickvergessen

File: lib/private/Collaboration/Collaborators/MailPlugin.php [hunk 1]
```diff
@@ -175,10 +174,13 @@ public function search($search, $limit, $offset, ISearchResult $searchResult) {
 
 		if (!$this->shareeEnumeration) {
 			$result['wide'] = [];
+			$userResults['wide'] = [];
```
Comment (nickvergessen) [inline lib/private/Collaboration/Collaborators/MailPlugin.php]: PS: also enumeration was possible before, pretty bad...

**Review by rullzer — APPROVED** [review]
Yes please!

**Comment by rullzer** [comment]
Ah failing tests

**Comment by nickvergessen** [comment]
Fixed

**Comment by blizzz** [comment]
test case for it possible?

**Comment by MorrisJobke** [comment]
> test case for it possible?

Still pending and should be done together with the Backport.

**Comment by nickvergessen** [comment]
Tests in https://github.com/nextcloud/server/pull/8710

**Comment by nickvergessen** [comment]
Backport in https://github.com/nextcloud/server/pull/8749

**Pull request merged**